### PR TITLE
hydran-309: Create and read mandates

### PR DESCRIPTION
**Pull request by martinhansson80**
Rewrote MandateEntity to utilize periods with unique constraints.

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/main/java/se/sundsvall/myrepresentative/service/Mapper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			.withSignatoryPartyId(mandate.grantorDetails().signatoryPartyId())
+			.withGrantee(mandate.granteeDetails().partyId())
+			.withName(mandate.grantorDetails().name())
+			.withActiveFrom(mandate.activeFrom())
+			.withInactiveAfter(mandate.inactiveAfter())
+			.withNamespace(namespace);
+	}
+
+	public MandateDetails toMandateDetails(final MandateEntity mandateEntity) {
+		return new MandateDetails(
```
Comment (josc04) [inline src/main/java/se/sundsvall/myrepresentative/service/Mapper.java:27]: Why not use the handy `@Builder`-annotation that you've added to the API record classes (`MandateDetailsBuilder.create()....build()`) 🙂?

File: src/main/java/se/sundsvall/myrepresentative/service/Mapper.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import se.sundsvall.myrepresentative.api.model.GranteeDetails;
+import se.sundsvall.myrepresentative.api.model.GrantorDetails;
+import se.sundsvall.myrepresentative.api.model.MandateDetails;
+import se.sundsvall.myrepresentative.api.model.MandateStatus;
+import se.sundsvall.myrepresentative.integration.db.entity.MandateEntity;
+
+@Component
+public class Mapper {
+
+	public MandateEntity toMandateEntity(final String municipalityId, final String namespace, final CreateMandate mandate) {
```
Comment (josc04) [inline src/main/java/se/sundsvall/myrepresentative/service/Mapper.java:14]: Use ofNullable in methods to make mapper more NPE resistant?

File: src/test/java/se/sundsvall/myrepresentative/api/model/CreateMandateTest.java [hunk 3]
```diff
@@ -24,24 +22,18 @@ void testBuilder() {
...
-			.withValidFrom(VALID_FROM)
-			.withValidTo(VALID_TO)
+			.withActiveFrom(ACTIVE_FROM)
+			.withInactiveAfter(INACTIVE_AFTER)
 			.build();
 
 		assertBean(mandate);
 	}
 
-	@Test
```
Comment (josc04) [inline src/test/java/se/sundsvall/myrepresentative/api/model/CreateMandateTest.java:34]: Any particular reason for removing this test?

File: src/test/java/se/sundsvall/myrepresentative/service/MapperTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+		assertThat(details.granteeDetails().partyId()).isEqualTo(GRANTEE_PARTY_ID);
+		assertThat(details.municipalityId()).isEqualTo(MUNICIPALITY_ID);
+		assertThat(details.namespace()).isEqualTo(NAMESPACE);
+		assertThat(details.activeFrom()).isEqualTo(ACTIVE_FROM);
+		assertThat(details.inactiveAfter()).isEqualTo(INACTIVE_AFTER);
+		assertThat(details.created()).isEqualTo(entity.getCreated());
+		assertThat(details.updated()).isEqualTo(entity.getUpdated());
+		assertThat(details.status()).isEqualTo(STATUS);
+	}
+}
```
Comment (josc04) [inline src/test/java/se/sundsvall/myrepresentative/service/MapperTest.java:90]: Add tests for null and empty beans?

File: src/main/java/se/sundsvall/myrepresentative/api/MandatesResource.java [hunk 5]
```diff
@@ -45,25 +47,39 @@
...
+				useReturnTypeSchema = true),
+			@ApiResponse(responseCode = "409",
+				description = "Conflict",
+				content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE,
+					schema = @Schema(implementation = Problem.class)))
 		})
 	@PostMapping(value = "/mandates", consumes = APPLICATION_JSON_VALUE, produces = ALL_VALUE)
 	ResponseEntity<Void> createMandate(
 		@Parameter(name = "municipalityId", description = "Municipality id", example = "2281") @ValidMunicipalityId @PathVariable final String municipalityId,
+		@Parameter(name = "namespace", description = "Namespace", example = "MY_NAMESPACE") @PathVariable final String namespace,
```
[the remark this answers is not in the document]
Comment (martinhansson80) [reply src/main/java/se/sundsvall/myrepresentative/api/MandatesResource.java:71]: That's a great idea, I'll fix

File: src/main/java/se/sundsvall/myrepresentative/config/DataIntegrityExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ * Handle duplicate entries in the database and return a proper Problem response.
+ */
+@ControllerAdvice
+public class DataIntegrityExceptionHandler implements ProblemHandling {
+
+	@ExceptionHandler(DataIntegrityViolationException.class)
+	public ResponseEntity<Problem> handleDuplicateEntry(DataIntegrityViolationException ex) {
+		return ResponseEntity.status(CONFLICT)
+			.body(Problem.builder()
+				.withTitle("Duplicate entry")
```
[the remark this answers is not in the document]
Comment (martinhansson80) [reply src/main/java/se/sundsvall/myrepresentative/config/DataIntegrityExceptionHandler.java:23]: Hm, I'll see if I can make it a little bit more narrow.

**Review by AndreSoderlund — APPROVED** [review]
